### DISCUSSION TITLE
CUDA error: CUBLAS_STATUS_INVALID_VALUE

### DISCUSSION THREAD

**Discussion by jdiazram** [JavierBZ/Tarea3-IIC3697#5] [Q&A] [answered]
Hola Javier
Al avanzar en la programacion de la tarea, ahora tengo otro error que no he podido resolver, haber si me puedes orientar:
Cuando trato de calcular la loss:

_RuntimeError: CUDA error: CUBLAS_STATUS_INVALID_VALUE when calling `cublasSgemm( handle, opa, opb, m, n, k, &alpha, a, lda, b, ldb, &beta, c, ldc)`_

He tratado de buscar la solución, no encontrado algo que funcione...que crees tu que puede? ser para orientar la solución.

Muchas gracias

**Comment by JavierBZ** [comment, marked as answer]
Por qué estas usando with torch.no_grad()? esa linea solo se usa cuando estas evaluando la red (en este caso cuando haces un paso en el ambiente usando la red), pero no cuando calculas la loss. Ahi estas desactivando todos los calculos de auto-diferenciación que son necesarios para calcular la loss y propagarla por la red. Esto no es la causa del error pero si te afectara para las lineas siguientes

En cuanto el error me parece que puede ser un error de dimensiones del input. Que dimensiones tiene tu `obs_t`?

También te pido si puedes sacar las lineas que vienen después de tu error para que no este pública en este foro el cálculo de loss

Quedo atento

**Reply by jdiazram** [reply]
Al usar obs_t.shape me arroja -> torch.Size([32, 4, 84, 84])

**Reply by JavierBZ** [reply]
Parece estar bien, y antes de hacer el paso _forward_ con `Q(obs_t)` lo pasaste al _device_, en este caso `'cuda'`?

**Reply by JavierBZ** [reply]
Me refiero a pasar el tensor al device con `obs_t.to(device)`

**Reply by jdiazram** [reply]
Si, incluso lo normalizo por 255...algo quizás antes este molestando...

**Reply by JavierBZ** [reply]
Hola si prefieres puedes enviarme el código completo al mail (jebisbal@uc.cl) para ver que parte puede estar fallando

**Reply by jdiazram** [reply]
Te pasate...muchas gracias...enviado...
